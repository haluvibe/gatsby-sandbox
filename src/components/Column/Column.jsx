import React from 'react';
import Layout from '../Layout/Layout';
import uuidv4 from 'uuid/v4';

export default ({children, settings}) => {
  const {className, ...other} = settings;
  return (
    <div className={`${className} l-grid-cell`} {...other}>
      { children && children.map(component => <Layout key={uuidv4()} {...component} />) }
    </div>
  )
}