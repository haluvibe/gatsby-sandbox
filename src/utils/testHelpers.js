export const mockSettings = {
  id: 'id',
  className: 'a-class another-class',
  style: {
    backgroundColor: 'red',
    fontSize: 16,
  },
};

export const mockChildren = [
  {
    settings: mockSettings,
    component: 'section',
  },
];

export const mockColumns = [
  {
    settings: mockSettings,
    component: 'column',
  },
];

export const mockData = {
  site: {
    siteMetadata: {
      title: 'Mock Title',
    },
  },
};

export const mockHtml = '<div>Hello test world</div>';
