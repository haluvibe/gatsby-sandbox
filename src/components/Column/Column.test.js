import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Column from './Column';
import { mockSettings, mockChildren } from '../../utils/testHelpers';
jest.mock('uuid/v4', () => jest.fn(() => '9999-9999'));

test('Column component should render as expected', () => {
  const component = shallow(
    <Column settings={mockSettings} children={mockChildren} />
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
