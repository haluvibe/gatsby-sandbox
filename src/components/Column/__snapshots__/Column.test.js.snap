// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column component should render as expected 1`] = `
<div
  className="a-class another-class l-grid-cell"
  id="id"
  style={
    Object {
      "backgroundColor": "red",
      "fontSize": 16,
    }
  }
>
  <Component
    component="section"
    key="9999-9999"
    settings={
      Object {
        "className": "a-class another-class",
        "id": "id",
        "style": Object {
          "backgroundColor": "red",
          "fontSize": 16,
        },
      }
    }
  />
</div>
`;
